product_of_all_other_numbers/product_of_all_other_numbers.py

```diff
@@ -2,10 +2,26 @@
 Input: a List of integers
 Returns: a List of integers
 '''
+
+from functools import reduce
+from operator import mul
+
 def product_of_all_other_numbers(arr):
     # Your code here
 
-    pass
+    # * Initialize empty array to hold return value
+    # * loop through array once for each value:
+    # *     start a current index pointer at 0
+    # *     slice a new value, excluding the current index
+    # *     preform math on new list values
+    # *     append value to return list
+    # * return list
+
+    products = [] * (len(arr) - 1)
+    for key, value in enumerate(arr):
+        temp = arr[:key] + arr[key + 1:]
+        products.append(reduce(mul, temp))
+    return products
```

Comment:
This is an extremely good implementation. Very pythonic, love the use of both `reduce` and `operator.mul`.

Notes:

**1:** You're not actually using `value` in the implementation at all, so you should probably indicate that by changing it from `value` to `_`, which is the conventional way to indicate that you don't need something in memory:

```python
def product_of_all_other_numbers(arr):
    products = [] * (len(arr) - 1)
    for key, _ in enumerate(arr):
        temp = arr[:key] + arr[key + 1:]
        products.append(reduce(mul, temp))
    return products
```

Given that you only care about the index (which is what's being passed to `key`) and not the value, this is actually one of those rare cases that `range` makes more sense than `enumerate`:

```python
def product_of_all_other_numbers(arr):
    products = [] * (len(arr) - 1)
    for index in range(len(arr)):
        temp = arr[:index] + arr[index + 1:]
        products.append(reduce(mul, temp))
    return products
```

**2:** Similar to my notes on your merge sort, it appears that line 20 isn't actually doing anything. And given that you're never accessing `products` using an index, you should likely just instantiate an empty array and not change your code to work with index access at this point. Giving us a final function that looks like this:

```python
def product_of_all_other_numbers(arr):
    products = []
    for index in range(len(arr)):
        temp = arr[:index] + arr[index + 1:]
        products.append(reduce(mul, temp))
    return products
```